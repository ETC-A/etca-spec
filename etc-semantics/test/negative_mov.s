;-mstrict
;0,16,64,128
    movx %rx0, -1
end:
    jmp end
