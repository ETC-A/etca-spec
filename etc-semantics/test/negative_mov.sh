set -eu
./scripts/test.sh "$1" "$2" "test/negative_mov.bin" 0
./scripts/test.sh "$1" "$2" "test/negative_mov.bin" 16
./scripts/test.sh "$1" "$2" "test/negative_mov.bin" 64
./scripts/test.sh "$1" "$2" "test/negative_mov.bin" 128
