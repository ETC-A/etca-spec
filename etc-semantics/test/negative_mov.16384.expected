<etc>
  <k>
    #halt "Halted successfully!" ~> .
  </k>
  <pc> 32770 </pc>
  <registers>
    0 |-> -1
    1 |-> 0
    2 |-> 0
    3 |-> 0
    4 |-> 0
    5 |-> 0
    6 |-> 0
    7 |-> 0
  </registers>
  <memory>
    128 |-> b"Y\x1f\x8e\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
  </memory>
  <flags>
    <carry>    false </carry>
    <zero>     false </zero>
    <negative> false </negative>
    <overflow> false </overflow>
  </flags>
  <machine-details>
    <cpuid1> 16384 </cpuid1>
    <cpuid2> 0 </cpuid2>
    <feat>   0 </feat>
    <reg-width> doubleword </reg-width>
    <reg-count> 8 </reg-count>
    <evil-mode> false </evil-mode>
  </machine-details>
  <in> </in>
  <out> </out>
</etc>
